firewood/src/merkle.rs

```diff
@@ -286,50 +286,60 @@ impl<T: ReadLinearStore> Merkle<T> {
 
     pub fn dump_node(
         &self,
-        addr: LinearAddress,
+        node: &Child,
         seen: &mut HashSet<LinearAddress>,
         writer: &mut dyn Write,
     ) -> Result<(), std::io::Error> {
-        let node = self.read_node(addr)?;
+        let (addr, hash) = match node {
+            Child::None => return Ok(()),
+            Child::Address(addr) => (addr, None),
+            Child::AddressWithHash(addr, hash) => (addr, Some(hash)),
+        };
+
+        let node = self.read_node(*addr)?;
+
         match &*node {
             Node::Branch(b) => {
-                write!(writer, "  {addr}[label=\"@{addr}")?;
+                write!(writer, "  {addr}[label=\"@{addr} hash={hash:?}")?;
```

Comment:
Seems like if `Child` implements `Debug` we can move all this logic there

Reply:
Done